plugins/scm.js

```diff
@@ -1,17 +1,20 @@
 'use strict';
 
+const core = require('../core');
 const Joi = require('joi');
 const models = require('../models');
-const core = require('../core');
-const scmUri = Joi.reach(models.pipeline.base, 'scmUri').required();
-const token = Joi.reach(models.user.base, 'token').required();
-const sha = Joi.reach(models.build.base, 'sha').required();
+
 const buildStatus = Joi.reach(models.build.base, 'status').required();
-const jobName = Joi.reach(models.job.base, 'name').optional();
-const username = Joi.reach(models.user.base, 'username').required();
 const checkoutUrl = Joi.reach(models.pipeline.create, 'checkoutUrl').required();
-const prNum = Joi.reach(core.scm.hook, 'prNum').allow(null).optional();
+const jobName = Joi.reach(models.job.base, 'name').optional();
 const pipelineId = Joi.reach(models.pipeline.base, 'id').optional();
+const prNum = Joi.reach(core.scm.hook, 'prNum').allow(null).optional();
+const scmContext = Joi.reach(models.pipeline.base, 'scmContext').required();
+const scmUri = Joi.reach(models.pipeline.base, 'scmUri').required();
+const scmUrl = Joi.reach(core.scm.repo, 'url').required();
```

Comment:
I forgot.. Fixed.

Screwdriver build should fail until merging https://github.com/screwdriver-cd/data-schema/pull/152 . Because `models.pipeline.base.scmContext` does not exist.